client.go

```diff
@@ -44,6 +44,20 @@ var (
 	respReadLimit = int64(4096)
 )
 
+type GoLogger interface {
```

Comment:
does it really need all these methods ? I would assume `retryablehttp` as a lib should provide only one level of logging with possible formatting. Then you could provide a supplementary `LogFunc` struct, 

```
client.Logger = retryablehttp.LogFunc(func(args ...interface{}){
      logrus.Println(args) 
// or 
//   logrus.Debugln(args)
})
``` 
and similar for formatted logging if needed. 

Interface would be defined as: 

```
// A Logger is a minimalistic interface for the SDK to log messages to. Should
type Logger interface {
	Log(...interface{})
}

```